articles/understanding-hill-climbing-in-ai/index.md

```diff
@@ -0,0 +1,114 @@
+* **No Backtracking:** A hill-climbing algorithm only works on the current state and succeeding states (future). It does not look at the previous states. 
+* **Feedback mechanism:** The algorithm has a feedback mechanism that helps it decide on the direction of movement (whether up or down the hill). The feedback mechanism is enhanced through the [generate-and-test technique](http://intelligence.worldofcomputing.net/ai-search/generate-and-test-search.html#:~:text=%20Algorithm%3A%20Generate-And-Test%20%201%201.Generate%20a%20possible,quit%20else%20go%20to%20step%201.%20More%20). 
+* **Incremental change:** The algorithm improves the current solution by incremental changes. 
+ 
+### State-Space Diagram Analysis
+A state-space diagram provides a graphical representation of states and the cost or objective functions. If the objective function is the y-axis, we aim to establish the local maximum and global maximum. If the cost function represents this axis, we aim to establish the local minimum and global minimum. More information about local minimum, local maximum, global minimum, and global maximum can be found [here](https://en.wikipedia.org/wiki/Maxima_and_minima). 
+
+The following diagram shows a simple state-space diagram. The objective function has been shown on the y-axis, while the state-space represents the x-axis.
+
+![Hill Climbing Algorithm in AI](/engineering-education/understanding-hill-climbing-in-ai/hill-climbing-algorithm-in-ai.png)
+
+[Image Source: Javat Point](https://static.javatpoint.com/tutorial/ai/images/hill-climbing-algorithm-in-ai.png)
+
+A state-space diagram consists of various regions that can be explained as follows;
+* **Local maximum:** A local maximum is a solution that surpasses other neighboring solutions or states but is not the best possible solution. 
+* **Global maximum:** This is the best possible solution achieved by the algorithm. 
+* **Current state:** This is the existing or present state.
+* **Flat local maximum:** This is a flat region where the neighboring solutions attain the same value. 
+* **Shoulder:** This is a plateau whose edge is stretching upwards. 
+
+### Problems in Hill Climbing 
+There are three regions in which a hill-climbing algorithm cannot attain a global maximum or the optimal solution: local maximum, ridge, and plateau. 
+
+#### Local maximum
+At this point, the neighboring states have lower values than the current state. The greedy approach feature will not move the algorithm to a worse off state. This will lead to the hill-climbing process's termination, even though this is not the best possible solution.
+
+This problem can be solved by searching for new paths to move from an undesirable state. This can be done through [backpropagation](https://en.wikipedia.org/wiki/Backpropagation). 

```

Comment:
Computing BPA is done via gradient descent usually. There exist other methods to find the optimum cost function as well. However, SGD also suffers from local maxima and minima. Can you check this statement once? I believe BPA also suffers from the local maximum. For example, momentum, Adaptive Gradients, and Adam are optimizers modified to combat this problem. Put across your views in a detailed manner.

Reply:
You are right. I have revised this part.